core/src/main/java/io/grpc/internal/ServiceConfigInterceptor.java

```diff
@@ -104,6 +115,7 @@ void handleUpdate(Map<String, Object> serviceConfig) {
     // Okay, service config is good, swap it.
     serviceMethodMap.set(Collections.unmodifiableMap(newServiceMethodConfigs));
     serviceMap.set(Collections.unmodifiableMap(newServiceConfigs));
+    nameResolveComplete = true;
```

Comment:
Is synchronization necessary for this?

Reply:
A note: If the stream listener is closed because of a status in a trailer from the server, `nameResolveComplete` will be observed to be true even not volatile.

Volatile is needed only for the case when the stream listener is closed for other reasons.